extensions-utils/src/dependencies/download_ic_binaries.rs

```diff
@@ -0,0 +1,64 @@
+use backoff::future::retry;
+use backoff::ExponentialBackoffBuilder;
+use flate2::read::GzDecoder;
+use std::path::Path;
+use std::time::Duration;
+use std::{fs, io::copy};
+use tokio::runtime::Runtime;
+
+pub fn download_ic_binary(replica_rev: &str, binary_name: &str, destination_path: &Path) {
+    let arch = match std::env::consts::ARCH {
+        "x86_64" => "x86_64",
+        "aarch64" => "x86_64", // let's rely on rosetta2 for now, since ic binaiers are not available for arm64
+        _ => panic!("Unsupported architecture"),
+    };
+    let os = match std::env::consts::OS {
+        "macos" => "darwin",
+        "linux" => "linux",
+        // "windows" => "windows", // unsupported till dfx supports windows
+        _ => panic!("Unsupported OS"),
+    };
+
+    let url = format!(
+        "https://download.dfinity.systems/ic/{replica_rev}/openssl-static-binaries/{arch}-{os}/{binary_name}.gz",
+        arch = arch,
+        os = os,
+        binary_name = binary_name,
+    );
+    println!("Downloading {}", url);
+
+    let bytes = Runtime::new().unwrap().block_on(download_bytes(&url));
+    let mut d = GzDecoder::new(&*bytes);
+    let tempdir = tempfile::tempdir().expect("Failed to create temp dir");
+    let temp_file = tempdir.path().join(binary_name);
+    let mut temp = fs::File::create(&temp_file).expect("Failed to create the file");
+    copy(&mut d, &mut temp).expect("Failed to copy content");
+
+    fs::rename(temp_file, &destination_path).expect("Failed to move extension");
+    #[cfg(unix)]
+    {
+        use std::os::unix::fs::PermissionsExt;
+        dfx_core::fs::set_permissions(&destination_path, std::fs::Permissions::from_mode(0o500))
+            .expect("Failed to set permissions");
+    }
```

Comment:
If we set the permission first, then we either get an executable downloaded binary in destination path, or nothing, but we never get a not-executable downloaded binary in the destination path.
```suggestion
    #[cfg(unix)]
    {
        use std::os::unix::fs::PermissionsExt;
        dfx_core::fs::set_permissions(&destination_path, std::fs::Permissions::from_mode(0o500))
            .expect("Failed to set permissions");
    }
    fs::rename(temp_file, &destination_path).expect("Failed to move extension");
```